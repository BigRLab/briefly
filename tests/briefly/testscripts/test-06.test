#> Branch/combine execution.

from briefly import *
from testutils import *

objs = Pipeline("test-06")
counter = 0

c = []
for n in range(5):
  if n == 0:
    c = [objs.phony()]
    continue

  nc = []
  for i in range(n + 1):
    counter += 1
    deps = []
    if i < n:
      deps.append(c[i])
    if i > 0:
      deps.append(c[i-1])
    nc.append(objs | test(counter, *deps))
  c = nc

for n in range(len(c) - 1, 0, -1):
  nc = []
  for i in range(n):
    counter += 1
    node = objs | test(counter, c[i], c[i+1])
    nc.append(node)
  c = nc

print len(c)
objs.run(*c)
