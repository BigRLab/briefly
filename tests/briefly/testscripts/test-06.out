>> 1 [True, True]
>> 10 [True, True]
>> 11 [True, True, True]
>> 12 [True, True, True]
>> 13 [True, True, True]
>> 14 [True, True]
>> 15 [True, True, True]
>> 16 [True, True, True]
>> 17 [True, True, True]
>> 18 [True, True, True]
>> 19 [True, True, True]
>> 2 [True, True]
>> 20 [True, True, True]
>> 21 [True, True, True]
>> 22 [True, True, True]
>> 23 [True, True, True]
>> 24 [True, True, True]
>> 3 [True, True]
>> 4 [True, True, True]
>> 5 [True, True]
>> 6 [True, True]
>> 7 [True, True, True]
>> 8 [True, True, True]
>> 9 [True, True]
